File: ideaproject/javaserializer/testsrc/com/houseelectrics/serializer/test/JSExecuteUtilJavascript.java
package com.houseelectrics.serializer.test;

/**
 * Created by roberttodd on 01/12/2014.
 */
public class JSExecuteUtilJavascript
{
    public final static String NewLine = "\r\n";
    public final static String ShowSourceFunctionName = "showjssource";
    public final static String ShowSourceFunction =
    "function " + ShowSourceFunctionName + "(f)" + NewLine +
    "{" + NewLine +
    "    var w = window.open('','','width=400, resizable=yes, scrollbars');" + NewLine +
    "    w.document.write('<pre>' + f + '</pre>');" + NewLine +
    "    w.document.close();" + NewLine +
    "    w.focus();" + NewLine +
    "};";
    public final static String AssertEqualFunction =
    "var assertEqual = function(expected, actual, message)" + NewLine +
    "{" + NewLine +
    "    if (expected!=actual)" + NewLine +
    "    {" + NewLine +
    "        throw message +  ' expected:' + expected + ' actual:' + actual;" + NewLine +
    "    }" + NewLine +
    "};";

    public static String getRunTest4TableRowFunction(String TablePageRunTestFunctionName, String crossimagesrc, String tickimagesrc, String questionmarkimagesrc)
    {
        String strJs = "function " + TablePageRunTestFunctionName + "(passimageid, testFunction, clear)" + NewLine +
        "{" + NewLine +
        "    var theimage = document.getElementById(passimageid);" + NewLine +
        "    if (clear)" + NewLine +
        "    {" + NewLine +
        "        theimage.src=runtest.questionmarkimagesrc;" + NewLine +
        "        return;" + NewLine +
        "    }" + NewLine +
        "    try" + NewLine +
        "    {" + NewLine +
        "        testFunction();" + NewLine +
        "        theimage.src=runtest.tickimagesrc;" + NewLine +
        "    }" + NewLine +
        "    catch (err)" + NewLine +
        "    {" + NewLine +
        "        theimage.src=runtest.crossimagesrc;" + NewLine +
        "    }" + NewLine +
        "}" + NewLine +
        "runtest.crossimagesrc='" + crossimagesrc + "';" + NewLine +
        "runtest.tickimagesrc='" + tickimagesrc + "';" + NewLine +
        "runtest.questionmarkimagesrc='" + questionmarkimagesrc + "';" + NewLine;

        return strJs;
    }

}

File: ideaproject/javaserializer/testsrc/com/houseelectrics/serializer/test/JSExecuteUtilJavascript.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.houseelectrics.serializer.test
{
    public class JSExecuteUtilJavascript
    {
        public const string ShowSourceFunctionName = "showjssource";
       public const string ShowSourceFunction =
           @"function " + ShowSourceFunctionName + @"(f)
{
var w = window.open('','','width=400, resizable=yes, scrollbars');
w.document.write('<pre>' + f + '</pre>');
w.document.close();
w.focus();
};";
       public const string AssertEqualFunction =
@"var assertEqual = function(expected, actual, message)
{
    if (expected!=actual)
       {
            throw message +  ' expected:' + expected + ' actual:' + actual;
       }
};";

       public static string getRunTest4TableRowFunction(string TablePageRunTestFunctionName, string crossimagesrc, string tickimagesrc, string questionmarkimagesrc) 
       {
return @"function " + TablePageRunTestFunctionName + @"(passimageid, testFunction, clear)
{
var theimage = document.getElementById(passimageid);
if (clear)
  {
     theimage.src=runtest.questionmarkimagesrc;
     return;
  }
try
{
testFunction();
theimage.src=runtest.tickimagesrc;
}
catch (err)
{
theimage.src=runtest.crossimagesrc;
}
}
runtest.crossimagesrc='" + crossimagesrc + @"';
runtest.tickimagesrc='" + tickimagesrc + @"';
runtest.questionmarkimagesrc='" + questionmarkimagesrc + @"';
";
      }

    }



}
